lib/Generate.js

```diff
@@ -190,39 +144,24 @@ function generateMainModule(
     moduleName: string,
     possiblyTests: Array<string>,
   }> */,
-  generatedSrc /*: string */,
+  mainModule /*: { moduleName: string, path: string } */,
   processes /*: number */
-) /*: string */ {
+) /*: void */ {
   const testFileBody = makeTestFileBody(
     testModules,
     makeOptsCode(fuzz, seed, report, testFileGlobs, testFilePaths, processes)
   );
 
-  // Generate a filename that incorporates the hash of file contents.
-  // This way, if you run e.g. `elm-test Foo.elm` and then `elm-test Bar.elm`
-  // and then re-run `elm-test Foo.elm` we still have a cached `Main` for
-  // `Foo.elm` (assuming none of its necessary imports have changed - and
-  // why would they?) so we don't have to recompile it.
-  const salt = Murmur.murmur3(testFileBody);
-  const moduleName = 'Main' + salt;
```

Comment:
The comment here says that the use case is:

```
elm-test Foo.elm
elm-test Bar.elm
# We should have a ready-to-go generated MainXXXX.elm file here:
elm-test Foo.elm
```

But that’s not true. There will be a unique MainXXXX.elm file for every invocation there. Why? Because the murmur hash is based on the MainXXXX.elm file contents (by design, according to the comment), and the file contents contain the _seed,_ which is randomly generated on startup. So the file contents _won’t_ be exactly the same on the second `elm-test Foo.elm` run, so it’ll get a new hash and a new generated file name and as such not benefit from any “cached” stuff.

What about watch mode, then? Was there any benefit from the hashing there? Because there you run with the same seed all the time, so the file contents shouldn’t change that often? That is all true – but removing the hash makes no difference. Before, you’d have Main3541244.elm for a bunch of runs, now you have just Main.elm for a bunch of runs.

The only time we _could_ reuse an old MainXXXX.elm is when you rerun tests with the `--seed` flag. But even then I don’t see the point – we still do all the compilation just as usual without skipping anything, and a comment below even mentions the importance of never skipping writing the generated .elm file to update the timestamp:

> // Always write the file, in order to update its timestamp. This is important,
> // because if we run `elm-make Main123456.elm` and that file's timestamp did
> // not change, elm-make will short-circuit and not recompile *anything* - even
> // if some of Main's dependencies (such as an individual test file) changed.

So I feel confident the hashing made no difference. I’ve not noticed any slowdown or anything from removing it. On the contrary, not hashing should be faster since we now do less work! And we get rid of one more dependency.

Another benefit is that `elm-stuff/` no longer grows indefinitely, with one new `MainXXXX.elm` file – along with .elmi and .elmo files – per `elm-test` invocation, basically. We now overwrite the same `Main.elm` all the time.

Reply:
Brilliant!